mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -606,11 +554,6 @@ impl MmCtxBuilder {
         self
     }
 
-    pub fn with_secp256k1_key_pair(mut self, key_pair: KeyPair) -> Self {
```

Comment:
To perform fewer changes, is it possible to keep `with_secp256k1_key_pair` and derive it separately?

Reply:
I think we should remove `ctx.secp256k1_key_pair` from the context. One of the reasons is MetaMask integration. There should be a module dedicated to storing the state of the private key policy.

Reply:
Probably, I didn't understand what you meant with
> keep with_secp256k1_key_pair and derive it separately?

Reply:
> Probably, I didn't understand what you meant with

I meant deriving a private key during initialization and passing it to `with_secp256k1_key_pair`.

> One of the reasons is MetaMask integration. 

As we discussed in DM, it's likely preferred to initialize MM2 "as usual" and only make another mode for ETH coins. Though, we need a confirmation from Tony and Yurii on this.

Reply:
I'm in favor of not storing the `secp256k1_key_pair` (even renaming it to `internal_secp256k1_key_pair`) at the `MmCtx`, because it can lead to incorrect use of it. If we refactor it now, and it will be more convenient to add new private key policies in the future.

Reply:
Ok, no problem 🙂